Fix -Wincompatible-pointer-types error in bundled sip with gcc 14.

Source: https://bugs.gentoo.org/show_bug.cgi?id=919000#c1
Upstream issue: https://github.com/wxWidgets/Phoenix/issues/2493

The fix is in upstream sip but the py3-wxpython release bundles an older
version at the moment.

Error:

```
../../../../sip/siplib/siplib.c: In function 'parsePass1':
../../../../sip/siplib/siplib.c:3900:20: error: assignment to 'sipSimpleWrapper
*' {aka 'struct _sipSimpleWrapper *'} from incompatible pointer type 'PyObject
*' {aka 'struct _object *'} [-Wincompatible-pointer-types]
 3900 |             *selfp = va_arg(va, PyObject *);
      |                    ^
```

--- wxPython-4.2.1-origin/sip/siplib/siplib.c
+++ wxPython-4.2.1/sip/siplib/siplib.c
@@ -3897,7 +3897,7 @@
     {
     case '#':
             /* A ctor has an argument with the /Transfer/ annotation. */
-            *selfp = va_arg(va, PyObject *);
+            *selfp = (sipSimpleWrapper *) va_arg(va, PyObject *);
             break;

     case 'B':
